### PR TITLE
adds internal id support in createDocumentl

### PR DESCRIPTION
**Pull request by fanatic75**

### CODE REVIEW

File: src/Database/Adapter/MariaDB.php [hunk 1]
```diff
@@ -612,12 +612,24 @@ public function createDocument(string $collection, Document $document): Document
             $bindIndex++;
         }
 
+        // Insert manual id if set
+        if (!empty($document->getInternalId())) {
```
[the remark this answers is not in the document]
Comment (fanatic75) [reply src/Database/Adapter/MariaDB.php:616]: I think we should not allow anyone to give internalId 0 as auto increment starts with 1 only. 
replacing empty to is_null won't be a good idea so adding another condition just to check if it's 0 can be done but I still think 0 shouldn'tbe allowed.

File: src/Database/Validator/Structure.php [hunk 2]
```diff
@@ -37,7 +37,7 @@ class Structure extends Validator
         [
             '$id' => '$internalId',
             'type' => Database::VAR_STRING,
-            'size' => 255,
+            'size' => 0,
```
[the remark this answers is not in the document]
Comment (fanatic75) [reply src/Database/Validator/Structure.php]: removed this change

File: tests/Database/Base.php [hunk 3]
```diff
@@ -933,6 +933,50 @@ public function testCreateDocument(): Document
...
+        $this->assertIsInt($manualIdDocument->getAttribute('bigint'));
+        $this->assertEquals(8589934592, $manualIdDocument->getAttribute('bigint'));
+        $this->assertIsFloat($manualIdDocument->getAttribute('float'));
+        $this->assertEquals(5.55, $manualIdDocument->getAttribute('float'));
+        $this->assertIsBool($manualIdDocument->getAttribute('boolean'));
+        $this->assertEquals(true, $manualIdDocument->getAttribute('boolean'));
+        $this->assertIsArray($manualIdDocument->getAttribute('colors'));
+        $this->assertEquals(['pink', 'green', 'blue'], $manualIdDocument->getAttribute('colors'));
+        $this->assertEquals([], $manualIdDocument->getAttribute('empty'));
+        $this->assertEquals('Works', $manualIdDocument->getAttribute('with-dash'));
```
Comment (abnegate) [inline tests/Database/Base.php:979]: Let's add a test to make sure that it's still set to the same after fetching it with `getDocument` as well

Comment (fanatic75) [reply tests/Database/Base.php:979]: added this

**Review by abnegate — APPROVED** [review]

**Pull request merged**